SpatialGDK/Source/SpatialGDKTests/Private/LocalDeploymentManager/LocalDeploymentManagerTest.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "LocalDeploymentManager.h"
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+#include "SpatialGDKDefaultWorkerJsonGenerator.h"
+#include "SpatialGDKEditorSettings.h"
+
+#include "CoreMinimal.h"
+
+#define LOCALDEPLOYMENT_TEST(TestName) \
+	TEST(Services, LocalDeployment, TestName)
+
+namespace
+{
+	static const double MAX_WAIT_TIME_FOR_LOCAL_DEPLOYMENT_OPERATION = 10.0;
+	enum class EDeploymentState { IsRunning, IsNotRunning };
+
+	FLocalDeploymentManager* GetLocalDeploymentManager()
+	{
+		FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+		FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+		return LocalDeploymentManager;
+	}
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND(StartDeployment);
+bool StartDeployment::Update()
+{
+	bool bRedeployRequired = false;
+	if (!GenerateDefaultWorkerJson(bRedeployRequired))
```

Comment:
Won't this reset any local settings on the machine?

Reply:
I'll fix that by passing a custom WorkerJsonPath variable to use.